### PR TITLE
[ Feature ] Abstract common entity

### PR DESCRIPTION
**Pull request by shift-developer**
### Pull request type
- Feature

### What is the current behavior?
-  No common module
- Validation pipe options in main.ts file
- No common base entity
### What is the new behavior?
- Common module structure created
- Validation pipe options decoupled and exceptionFactory added with improved validation errors adding the dto field as key
- Path alias created for common
- Base common abstract entity created with parts such as id column, uuid, created_at, from which the other models will inherit.
### Tasks
- [x] Create common module structure
- [x] Create path alias form common
- [x] Create utils in common file with decoupled pipe validation options
- [x] Create entities folder with base entity

### CODE REVIEW

File: src/common/common.module.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+import { Module } from '@nestjs/common';
```
Comment (slasia) [inline src/common/common.module.ts]: it is not necessary to create a new module for the common files, you can create a folder that is `/common `or `/shared` and there put everything that is shared between several modules

**Review by slasia — CHANGES_REQUESTED** [review]
Good work! I left a comment

File: src/common/common.module.ts (hunk 1 above)
Comment (shift-developer) [reply src/common/common.module.ts]: Thanks Seba! I agree with you

File: src/common/entities/abstract.entity.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  DeleteDateColumn,
+  Generated,
+  PrimaryGeneratedColumn,
+  UpdateDateColumn,
+} from 'typeorm';
+
+export abstract class AbstractEntity extends BaseEntity {
+  @PrimaryGeneratedColumn()
+  @Exclude()
+  public id: number;
```
Comment (slasia) [inline src/common/entities/abstract.entity.ts:15]: tip for the future: You can use uuid to the id column

File: src/home/home.module.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { Module } from '@nestjs/common';
+import { HomeService } from './home.service';
```
Comment (slasia) [inline src/home/home.module.ts]: You can use path alias

**Review by slasia — APPROVED** [review]
LGTM

File: src/common/entities/abstract.entity.ts (hunk 2 above)
Comment (shift-developer) [reply src/common/entities/abstract.entity.ts:15]: Thanks Seba!!

File: src/home/home.module.ts (hunk 3 above)
Comment (shift-developer) [reply src/home/home.module.ts]: Yep, I'll fix it. Thanks

**Pull request merged**